sdk/core/azure-core/src/main/java/com/azure/android/core/http/interceptors/ScrubEtagInterceptor.java

```diff
@@ -0,0 +1,40 @@
+package com.azure.android.core.http.interceptors;
+
+import androidx.annotation.NonNull;
+
+import java.io.IOException;
+
+import okhttp3.Interceptor;
+import okhttp3.Response;
+
+/**
+ * Wraps any potential error responses from the service and applies post processing of the response's eTag header to
+ * standardize the value.
+ */
+public class ScrubEtagInterceptor implements Interceptor {
+    private static final String ETAG = "ETag";
```

Comment:
This differs from the Java implementation, there they use the value `eTag`. Any reason for the change?

Reply:
I'm not sure why the Java SDK is using "eTag", but since the HTTP [specification](https://tools.ietf.org/html/rfc7232#section-2.3) states that the name of the header is "ETag" and all examples I've seen online use it as well, I decided to stick to this casing.

Reply:
I'm surprised that OkHTTP doesn't seem to have a standardized enum or class that contains the header names. So I'm good with this change.

Reply:
Just checked and it seems that when you try to get a header value the case does not matter, however, I would still prefer to follow the specification.